### ISSUE TITLE
Comment: Recipe: Basic OAuth (guides/auth/recipe.oauth-basic.md)

### ISSUE THREAD

**Issue by samburgers** [feathersjs/docs#1078]
This creates duplicate entries in the `users` table upon each subsequent authentication/login. Same issue with other Oauth strategies (tested with facebook). Is this by design, am i missing something?

**Comment by edwardsmarkf** [comment]
hi - i am working on the same thing.

for fun**, i created a new feathers-app using gmail-oath2 and
sequelize-mariadb.  there was only one entry in the mariadb users table,
after logging out and back in a couple of times.


what is your data storage?  NeDB?

if you want, write me back at mark (at) edwardsmark (dot) com



** (yes its a strange concept of fun)

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>


Thank you,

Mark Edwards

**Comment by samburgers** [comment]
Thanks for the reply! Yes, NeDB at the moment (during development), do you think the issue would be with that?

**Comment by edwardsmarkf** [comment]
>>>  do you think the issue would be with that?

yes!   i don't profess to be any sort of NeDB expert (unless you call doing
" cat ./data/users.db;' to be an expert)  but i noticed the same thing.
you can go a bit further and do insert/delete on another service and
observe the same behavior.

FYI - the oauth2 examples are sorta lacking, imho.  i put together a series
of gist's that really do everything for you, if you are interested.  you
just create a play directory and run them.  you might be particularly
interested in retrieving the email address *AND the person's name* from
their profile as well.  i did this in both gmail and github-login.  no
facebook login yet.




<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link>
<#m_4351834203635512858_DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>


Thank you,

Mark Edwards

**Comment by samburgers** [comment]
Ok, early tests confirm that the default CLI scaffolding with Mongo as the DB doesn't have this issue.

**Comment by daffl** [comment]
The standard query the oAuth2 verifier is using can be found at https://github.com/feathersjs/authentication-oauth2/blob/master/lib/verifier.js#L74 - it's odd that this seems to make a difference between NeDB and MongoDB

**Comment by samburgers** [comment]
I think it might be to do with the NeDB model schema/validation that the CLI generates when selecting to use NeDB. Not sure if it ensure `uniques` for OAuth users (only for email addresses). I could be completely wrong tho.

**Comment by edwardsmarkf** [comment]
hey sam -

i added email, familyName, givenName to the user model, and created a user
hook - results below:

MariaDB [Users]> select  *  from users;
+----+-----------------------+----------------------+--------------+------------+-----------+---------------------+---------------------+
| id | googleId              | email                | displayName  |
familyName | givenName | createdAt           | updatedAt           |
+----+-----------------------+----------------------+--------------+------------+-----------+---------------------+---------------------+
|  1 | 114113878629589347134 | mark@edwardsmark.com | Mark Edwards |
Edwards    | Mark      | 2018-02-17 15:47:35 | 2018-02-17 15:47:35 |
+----+-----------------------+----------------------+--------------+------------+-----------+---------------------+---------------------+
1 row in set (0.00 sec)



<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>


Thank you,

Mark Edwards

**Comment by samburgers** [comment]
Mongo + Mongoose has been working well and without issue. Here is my edited `/models/user.models.js` to accept facebook auth if anyone finds it useful.

```
// users-model.js - A mongoose model
//
// See http://mongoosejs.com/docs/models.html
// for more of what you can do here.
module.exports = function(app) {
  const mongooseClient = app.get('mongooseClient');
  const users = new mongooseClient.Schema(
    {
      email: {
        type: String,
        unique: true
      },

      password: {
        type: String
      },

      facebook: {
        profile: {
          id: String,
          email: String,
          displayName: String,
          gender: String,
          profileUrl: String,
          emails: Array,
          photos: Array
        }
      }
    },
    {
      timestamps: true
    }
  );

  return mongooseClient.model('users', users);
};
```

**Comment by daffl** [comment]
So this problem is specific to NeDB right?

**Comment by samburgers** [comment]
It seems so yes
